spark/v3.4/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestBranchDDL.java

```diff
@@ -89,6 +91,13 @@ public void testCreateBranch() throws NoSuchTableException {
         () -> sql("ALTER TABLE %s CREATE BRANCH %s", tableName, branchName));
   }
 
+  @Test
+  public void testCreateBranchOnEmptyTable() {
+    Assertions.assertThatThrownBy(() -> sql("ALTER TABLE %s CREATE BRANCH %s", tableName, "b1"))
+        .isInstanceOf(ValidationException.class)
+        .hasMessageContaining("Cannot set b1 to unknown snapshot");
```

Comment:
Seems like we're relying on the throw from here? https://github.com/apache/iceberg/blob/master/core/src/main/java/org/apache/iceberg/TableMetadata.java#L1170

I'm not sure that's what we want to do, that's supposed to be for assigning a ref to an invalid snapshot. In this case we want to signal to a user that no explicit snapshot was specified and there is no current main so the `CREATE BRANCH` or `TAG` is invalid. Could we just explicitly throw based off that?

Reply:
discussed offline, looks like this is relate to implicit scala Long to primitive java.lang.long conversion, seems fragile and confusion if we rely on that to guard against bad input. So we ended check snapshotId not null before pass to java code and also make unit test more self-contained. 

@amogh-jahagirdar ready for another look, if new exception and error message looks good to you, I'll backport fix to other spark version and try to get it merged for 1.3.0 release